assets/src/edit-story/components/panels/design/preset/utils.js

```diff
@@ -129,41 +129,47 @@ function getTextInlineStyles(content) {
   };
 }
 
+function getUniquePresets(presets) {
+  const list = presets.map((preset) => JSON.stringify(preset));
+  return Array.from(new Set(list)).map((preset) => JSON.parse(preset));
+}
```

Comment:
Uh, this sounds a bit expensive. Are we sure it's performant? How often does this function run?

Reply:
Hm, can't really come up with a better solution though. But I really think we should performance test this with e.g. 50 objects and run it a couple of times a second and see if performance degrades.

Reply:
This only runs when new presets are being added from elements (user clicks on the + sign while elements selected). Unlikely that someone will add 50 elements with different presets to add those at the same time. Could happen in theory 🤷  Thoughts?

Reply:
Tested it out by adding tons of shapes (didn't really count how many but a lot) and visually it worked without any lag at all. From the profiler it looked like the click handler took almost the same time in the case of 2 elements vs 40+ elements (around 100-150ms).